flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/v2/FlinkSink.java

```diff
@@ -0,0 +1,616 @@
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.types.Row;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DistributionMode;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.PartitionField;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SerializableTable;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.common.DynFields;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
+import org.apache.iceberg.flink.FlinkWriteConf;
+import org.apache.iceberg.flink.FlinkWriteOptions;
+import org.apache.iceberg.flink.TableLoader;
+import org.apache.iceberg.flink.sink.EqualityFieldKeySelector;
+import org.apache.iceberg.flink.sink.PartitionKeySelector;
+import org.apache.iceberg.flink.sink.RowDataTaskWriterFactory;
+import org.apache.iceberg.flink.util.FlinkCompatibilityUtil;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.types.TypeUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.iceberg.TableProperties.WRITE_DISTRIBUTION_MODE;
+
+/**
+ * Flink v2 sink offer different hooks to insert custom topologies into the sink. It includes writers that can write
+ * data to files in parallel and route commit info globally to one Committer.
+ * Post commit topology will take of compacting the already written files and updating the file log after the
+ * compaction.
+ *
+ * <pre>{@code
+ *                            Flink sink
+ *               +------------------------------------------------------------------+
+ *               |                                                                  |
+ * +-------+     | +-------------+       +---------------+                          |
+ * | Map 1 | ==> | | writer 1 | =|       | committer 1   |                          |
+ * +-------+     | +-------------+       +---------------+                          |
+ *               |                 \                                                |
+ *               |                  \                                               |
+ *               |                   \                                              |
+ * +-------+     | +-------------+    \  +---------------+        +---------------+ |
+ * | Map 2 | ==> | | writer 2 | =| ---  >| committer 2   |  --->  | post commit   | |
+ * +-------+     | +-------------+       +---------------+        +---------------+ |
+ *               |                                                                  |
+ *               +------------------------------------------------------------------+
+ * }</pre>
+ */
+public class FlinkSink implements StatefulSink<RowData, StreamWriterState>,
```

Comment:
I would suggest not using `v2` in the package name, instead we can name the new sink impl as `IcebergSink`. That is also consistent with the FLIP-27 source `IcebergSource` vs the current `FlinkSource`

Reply:
done. rename to `IcebergSink`. thx.